l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,478 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {EfficientCall} from "@matterlabs/zksync-contracts/l2/system-contracts/libraries/EfficientCall.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @title ConsensusRegistry
+/// @dev Manages consensus nodes and committees for the L2 consensus protocol,
+/// owned by a governance protocol. Nodes act as both validators and attesters,
+/// each playing a distinct role in the consensus process. This contract facilitates
+/// the rotation of validator and attester committees, which represent a subset of nodes
+/// expected to actively participate in the consensus process during a specific time window.
+contract ConsensusRegistry is Ownable2Step {
+    // An array to keep track of node owners.
+    address[] public nodeOwners;
+    // A map of node owners => nodes.
+    mapping(address => Node) public nodes;
+
+    uint256 validatorsCommit;
+    uint256 attestersCommit;
+    PendingNodeRemoval[] pendingRemovals;
+
+    struct Node {
```

Comment:
Please make `IConsensusRegistry.sol` file with interface for the contract. Then you can declare `structs`/`events`/`error` there to make code a bit cleaner. 

This is similar to the [IGovernance](https://github.com/matter-labs/era-contracts/blob/main/l1-contracts/contracts/governance/IGovernance.sol) and [Governance](https://github.com/matter-labs/era-contracts/blob/main/l1-contracts/contracts/governance/Governance.sol#L20) separation.

Reply:
Added here: https://github.com/matter-labs/era-contracts/pull/555/commits/f90aa998d42f8cbb3235536e71f74e1e4c03064c

Plz also review the location.